### PR TITLE
expand the regex to find commands

### PR DESCRIPTION
**Pull request by 927589452**

### CODE REVIEW

**Comment by 927589452** [comment]
this is missing the 
```
\NewDocumentEnvironment
\RenewDocumentEnvironment
\ProvideDocumentEnvironment
\DeclareDocumentEnvironment
```
yet

**Comment by lervag** [comment]
Could you please add a description of this as well? And preferably also a test document to show the effect.

**Comment by 927589452** [comment]
Whats the best way to test the autocompletion?
Will open a feature request so the testfiles have a home

**Comment by lervag** [comment]
> Will open a feature request so the testfiles have a home

Actually, you could just as well add files under `issues/1488`. That is, no reason to open a new issue. Instead, you could just have added the text you put in #1490 to the original post here.

No harm done, though. Just my two cents on making this more concise.

> Whats the best way to test the autocompletion?

Good question. I typically do something like [the test files for #1479](https://github.com/lervag/vimtex/tree/master/test/issues/1479). It includes a Makefile, so testing is simply a `make` away.

(In the future, I plan to remove the use of Vader for auto testing, and instead rely on something similar to this type of Makefile based testing.)

**Comment by lervag** [comment]
I've updated the test files, please pull changes to your local version. I've added a Makefile and updated the minimal.vim file so that you can easily test. Note, right now, the test fails, because the command completion lacks a couple of the candidates like `VimtexStandardProvide`.

File: autoload/vimtex/complete.vim [hunk 1]
```diff
@@ -4,7 +4,7 @@
 " Email:      karl.yngve@gmail.com
 "
 
-function! vimtex#complete#init_buffer() abort " {{{1
+function! vimtex#complete#init_buffer() abort " {{{2
```
Comment (lervag) [inline autoload/vimtex/complete.vim]: I assume this was a mistake?

File: autoload/vimtex/complete.vim [hunk 2]
```diff
@@ -854,9 +854,9 @@ function! s:completer_env.gather_candidates_from_newenvironments() dict abort "
 
   let l:candidates = vimtex#parser#tex(b:vimtex.tex, {'detailed' : 0})
 
-  call filter(l:candidates, 'v:val =~# ''\v\\((re)?newenvironment|(New|Renew|Provide|Declare)DocumentEnvironment)''')
+  call filter(l:candidates, 'v:val =~# ''\v\\((renew|new)environment|(New|Renew|Provide|Declare)DocumentEnvironment)''')
   call map(l:candidates, '{
-        \ ''word'' : matchstr(v:val, ''\v\\((re)?newenvironment|(New|Renew|Provide|Declare)DocumentEnvironment)\*?\{\\?\zs[^}]*''),
+        \ ''word'' : matchstr(v:val, ''\v\\((renew|new)environment|(New|Renew|Provide|Declare)DocumentEnvironment)\*?\{\\?\zs[^}]*''),
```
Comment (lervag) [inline autoload/vimtex/complete.vim]: This change seems unnecessary.

**Comment by lervag** [comment]
See my two comments to the code. Also, just because I'm curious: From which package or class do this family of commands come from, e.g. `\NewDocumentEnvironment`, `\RenewDocumentEnvironment`, `\ProvideDocumentEnvironment`, and `\DeclareDocumentEnvironment`?

**Comment by 927589452** [comment]
> See my two comments to the code. Also, just because I'm curious: From which package or class do this family of commands come from, e.g. `\NewDocumentEnvironment`, `\RenewDocumentEnvironment`, `\ProvideDocumentEnvironment`, and `\DeclareDocumentEnvironment`?

that is all xparse

File: autoload/vimtex/complete.vim (hunk 2 above)
Comment (927589452) [reply autoload/vimtex/complete.vim]: it technically is, but this keeps the notion similiar for environment and command
this gives IMO a better readability and is needed to differentiate between `provide` and `renew`

I have no idea on performance, but I thought about going back to the beginning and splitting the `xparse` parsing into a dedicated function

and add some more for other packages

File: autoload/vimtex/complete.vim (hunk 1 above)
Comment (927589452) [reply autoload/vimtex/complete.vim]: probably, I will look into it, but monday is print time

**Comment by lervag** [comment]
Ok, thanks. Is this PR ready to merge, in your opinion?

**Comment by 927589452** [comment]
> Ok, thanks. Is this PR ready to merge, in your opinion?

depends on wether later regexes should
- extend these line
- add a function per package providing commands

**Comment by lervag** [comment]
I would keep it as it is for now, then consider to update later if necessary. For now, there is only a single package that adds to the standard commands (`xparse`).

**Comment by lervag** [comment]
Ok, I've merged this now. Thanks!

**Pull request closed without merging**